cmd/relayer/setup/setup.go

```diff
@@ -268,15 +271,92 @@ func getPreRunInfo(home string) (*consts.RollappData, *consts.HubData, string, e
 		pterm.Error.Println("failed to retrieve rollapp rpc endpoint: ", err)
 		return nil, nil, "", err
 	}
-	raRpc = strings.TrimSuffix(raRpc, "/")
+	pterm.Info.Printf("validating rollapp rpc endpoint: %s\n", raRpc)
+	validatedRpc, err := validateRollappRPCEndpoint(raID, raRpc)
+	if err != nil {
+		pterm.Error.Println("rollapp rpc endpoint validation failed:", err)
+		return nil, nil, "", err
+	}
 
 	raData := consts.RollappData{
 		ID:     raID,
-		RpcUrl: raRpc,
+		RpcUrl: validatedRpc,
 	}
 	return &raData, hd, kb, nil
 }
 
+type tendermintStatusResponse struct {
+	Result struct {
+		NodeInfo struct {
+			Network string `json:"network"`
+		} `json:"node_info"`
+	} `json:"result"`
+}
+
+func validateRollappRPCEndpoint(rollappID, endpoint string) (string, error) {
```

Comment:
so this checks that the on chain rpc endpoint, provided by the sequencer is actually pointing to the correct chain id?

move it to `utils/sequencer`, other then that - lgtm

Reply:
yea thx @artemijspavlovs . I pushed a fix